tests/e2e/scenarios/test_provider_capability_inventory.py

```
@@ -194,3 +202,272 @@ def test_tested_capabilities_have_executable_evidence_at_the_correct_seam():
+
+
+@pytest.mark.parametrize(
+    ("label", "call"),
+    [
+        ("direct", "    await _assert_readback(world)\n"),
+        # Matched via ast.Attribute, which is receiver-agnostic on purpose:
+        # a helper reached through a module or holder object still counts.
+        ("through a receiver", "    await helpers._assert_readback(world)\n"),
+    ],
+)
+def test_journey_evidence_accepts_a_genuinely_invoked_helper(label: str, call: str):
```

Ruff (0.15.21)

[warning] Unused function argument: `label`

(ARG001)